safenode/src/protocol/mod.rs

```diff
@@ -10,5 +10,102 @@
 pub mod error;
 /// Messages types
 pub mod messages;
+
+use self::error::{Error, Result};
+
+use libp2p::{
+    kad::{kbucket::Distance, KBucketKey as Key},
+    PeerId,
+};
+use serde::{Deserialize, Serialize};
+use xor_name::{XorName, XOR_NAME_LEN};
+
+/// This is the key in the network by which proximity/distance
+/// to other items (wether nodes or data chunks) are calculated.
+///
+/// This is the mapping from the XOR name used
+/// by for example self encryption, or the libp2p `PeerId`,
+/// to the key used in the Kademlia DHT.
+/// All our xorname calculations shall be replaced with the `KBucketKey` calculations,
+/// for getting proximity/distance to other items (wether nodes or data).
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize, Debug)]
+pub enum NetworkKey {
+    /// The NetworkKey is representing a PeerId.
+    PeerId(Vec<u8>),
+    /// The NetworkKey is representing an XorName of some data.
```

Comment:
I wonder if here we should include the types of data instead of just the generic XorAddr?